File: dart/lib/src/utils/check.dart
import '../errors/sns_errors.dart';

/// Checks a boolean condition and throws a specified error if the condition is false.
/// This function is intended for internal use only.
///
/// [condition] - The boolean condition to check.
/// [error] - The error to be thrown if the condition is false.
///
/// Throws the specified error if the condition is false.
void check(bool condition, SnsError error) {
  if (!condition) {
    throw error;
  }
}

File: js/src/utils/check.ts
import { SNSError } from "../error";

/**
 * Throws `error` unless `bool` is true, preserving its concrete SNS error type.
 *
 * @param bool Condition that must be true
 * @param error Error to throw when `bool` is false
 * @throws The supplied error when `bool` is false
 *
 * @example
 * ```ts
 * check(value !== undefined, new InvalidInputError("A value is required"));
 * ```
 */
export const check = <T extends SNSError>(bool: boolean, error: T) => {
  if (!bool) {
    throw error;
  }
};
